packages/core/streams/langchain-stream.test.ts

```diff
@@ -0,0 +1,139 @@
+import { ChatOpenAI } from 'langchain/chat_models/openai';
+import { HumanMessage } from 'langchain/schema';
+import { createClient } from '../tests/utils/mock-client';
+import { setup } from '../tests/utils/mock-service';
+
+describe('LangchainStream', () => {
+  if (typeof Response === 'undefined') {
+    xit("should skip this test on Node 16 because it doesn't support `Response`", () => {});
+    return;
+  }
+
+  let server: ReturnType<typeof setup>;
+  beforeAll(() => {
+    server = setup();
+  });
+  afterAll(() => {
+    server.teardown();
+  });
+
+  const { LangChainStream, StreamingTextResponse, experimental_StreamData } =
+    require('.') as typeof import('.');
+
+  function readAllChunks(response: Response) {
+    return createClient(response).readAll();
+  }
+
+  it('should be able to parse SSE and receive the streamed response', async () => {
+    const { stream, handlers } = LangChainStream();
+
+    const llm = new ChatOpenAI({
+      streaming: true,
+      openAIApiKey: 'fake',
+      configuration: {
+        baseURL: server.api,
+        defaultHeaders: {
+          'x-mock-service': 'openai',
+          'x-mock-type': 'chat',
+          'x-flush-delay': '5',
+        },
+      },
+    });
+
+    llm.call([new HumanMessage('hello')], {}, [handlers]).catch(console.error);
+
+    const response = new StreamingTextResponse(stream);
+
+    expect(await readAllChunks(response)).toEqual([
+      '',
```

Comment:
With Langchain, there is an empty text chunk at the beginning and the end. Is this desired?

Reply:
No, but it shouldn't be a problem and not worth investigating until we finalize the complex mode API imo